### PR TITLE
bind to second service

### PR DESCRIPTION
**Pull request by e-bunityants-exceed**

### CODE REVIEW

File: src/department/department.service.ts [hunk 1]
```diff
@@ -3,22 +3,141 @@ import { Injectable, Inject } from '@nestjs/common';
...
+    data.developers = resultArrDev.unique;
+
+    // deleted not exist child
+    if (data.hasOwnProperty(Entity.Team)) {
+      delete data[Entity.Team];
+    }
+
+    const notExist = await this.bindService.isChildExist(data);
+    if (data.hasOwnProperty(Entity.Unit)) {
+      data.unit = data.unit.filter((el) => notExist.indexOf(el) == -1);
```
Comment (iminyukov) [inline src/department/department.service.ts:40]: find instead

File: src/developer/developer.controller.ts [hunk 2]
```diff
@@ -30,7 +31,18 @@ export class DeveloperController {
...
-      return await this.developerService.create(data);
+      const dev = await this.developerService.create(data);
+      if (data.hasOwnProperty(Entity.Team)) {
+        this.bindService.setChild(dev.team, Entity.Developers, dev._id, Entity.Team);
+      } else if (data.hasOwnProperty(Entity.Unit)) {
+        this.bindService.setChild(dev.unit, Entity.Developers, dev._id, Entity.Unit);
+      } else {
+        this.bindService.setChild(dev.department, Entity.Developers, dev._id, Entity.Department);
+      }
+
```
Comment (iminyukov) [inline src/developer/developer.controller.ts:42]: const dev = await this.developerService.create(data);
if (data.hasOwnProperty(Entity.Team)) {
  this.bindService.setChild(dev.team, Entity.Developers, dev._id, Entity.Team);
} else if (data.hasOwnProperty(Entity.Unit)) {
  this.bindService.setChild(dev.unit, Entity.Developers, dev._id, Entity.Unit);
} else {
  this.bindService.setChild(dev.department, Entity.Developers, dev._id, Entity.Department);
}

for (let entity in data) {
  this.bindService.setChild(dev[entity], Entity.Developers, dev._id, Entity[entity]);
}

File: src/bind/bind.service.ts [hunk 3]
```diff
@@ -46,7 +53,7 @@ export class BindService {
     const uniqueArr = [...new Set(arr)];
```
Comment (d-anistratenko) [inline src/bind/bind.service.ts:53]: Naming

Comment (d-anistratenko) [inline src/bind/bind.service.ts:53]: Naming

File: src/bind/bind.service.ts [hunk 4]
```diff
@@ -108,34 +143,37 @@ export class BindService {
...
+        if (entity) {
+          existingIds.push(entity._id);
         }
       }
     };
 
-    if (item.hasOwnProperty(Entity.Developers) && isArray(item.developers)) {
-      const promiseArr = item.developers.map((el) =>
-        this.DeveloperModel.findById(el),
+    if (collection == Entity.Developers && isArray(item.developers)) {
```
Comment (d-anistratenko) [inline src/bind/bind.service.ts:156]: Best practice?

File: src/department/department.controller.ts [hunk 5]
```diff
@@ -1,29 +1,81 @@
...
-      unit: [],
-    });
-    // RMQ publish
-    this.departmentServ.send('create', dep);
-    return dep;
+  async create(@Body() data: DepartmentDto) {
+    try {
+      this.bindService.isValideArrayId(data, Entity.Unit);
+
+      const r = await this.departmentService.create(data);
```
Comment (d-anistratenko) [inline src/department/department.controller.ts:36]: "r"?

File: src/department/department.controller.ts [hunk 6]
```diff
@@ -1,29 +1,81 @@
...
+    try {
+      // validation Id
+      this.bindService.isId(id);
+      if (data[Entity.Unit]) {
+        this.bindService.isValideArrayId(data, Entity.Unit);
+      }
+      if (data[Entity.Developers]) {
+        this.bindService.isValideArrayId(data, Entity.Developers);
+      }
+      const upt = await this.departmentService.update(id, data);
```
Comment (d-anistratenko) [inline src/department/department.controller.ts:55]: upt?

File: src/department/department.service.ts [hunk 7]
```diff
@@ -3,22 +3,122 @@ import { Injectable, Inject } from '@nestjs/common';
...
+  async getAll() {
+    return await this.DepartmentModel.find({});
   }
 
-  async getAll(): Promise<any> {
-    return await this.DepartmentModel.find({});
+  async create(data: DepartmentDto) {
+    const res = {};
+    // filter unique id (return {unique:[], noUnique:[] })
+    const resultArrUnit = this.bindService.isUniqueArrId(data.unit);
```
Comment (d-anistratenko) [inline src/department/department.service.ts:27]: Rename all variables containing "Arr"

File: src/developer/developer.controller.ts [hunk 8]
```diff
@@ -40,7 +67,9 @@ export class DeveloperController {
   async update(@Param('id') id: string, @Body() data: DeveloperUpdateDto) {
     try {
       this.bindService.isId(id);
-      return await this.developerService.update(id, data);
+      const upt = await this.developerService.update(id, data);
```
Comment (d-anistratenko) [inline src/developer/developer.controller.ts:70]: upt?

File: src/team/team.service.ts [hunk 9]
```diff
@@ -34,16 +38,16 @@ export class TeamService {
...
         id: resultArr.noUnique,
       };
     }
-    if (notExistDev.length) {
-      res['notExist'] = {
-        massage: "This Developers does't exist!",
-        id: notExistDev,
-      };
-    }
+    // if (notExistDev.length) {
```
Comment (d-anistratenko) [inline src/team/team.service.ts]: Delete unused comments

File: src/unit/unit.controller.ts [hunk 10]
```diff
@@ -27,11 +38,18 @@ export class UnitController {
...
       // validation Id
       this.bindService.isId(id);
+
       if (data[Entity.Team]) {
         this.bindService.isValideArrayId(data, Entity.Team);
       }
+      if (data[Entity.Developers]) {
+        this.bindService.isValideArrayId(data, Entity.Developers);
+      }
+      const upt = await this.unitService.update(id, data);
```
Comment (d-anistratenko) [inline src/unit/unit.controller.ts:48]: upt?

File: src/unit/unit.service.ts [hunk 11]
```diff
@@ -125,12 +118,12 @@ export class UnitService {
         id: resultArrDev.noUnique,
       };
     }
-    if (notExist.length) {
-      res['notExist'] = {
-        massage: "This Id doesn't exist!",
-        id: notExist,
-      };
-    }
+    // if (notExist.length) {
```
Comment (d-anistratenko) [inline src/unit/unit.service.ts]: Delete unused comments

**Pull request merged**